src/npm.rs

```diff
@@ -31,3 +31,42 @@ pub fn npm_publish(path: &str) -> Result<(), Error> {
         Ok(())
     }
 }
+
+pub fn npm_login(
+    registry: Option<String>,
+    scope: Option<String>,
+    always_auth: bool,
+    auth_type: Option<String>,
+) -> Result<(), Error> {
+    let mut args = String::new();
+
+    if let Some(registry) = registry {
+        args.push_str(&format!(" --registry={}", registry));
+    }
+    if let Some(scope) = scope {
+        args.push_str(&format!(" --scope={}", scope));
+    }
+
+    if always_auth == true {
+        args.push_str(" --always_auth");
+    }
+
+    if let Some(auth_type) = auth_type {
+        args.push_str(&format!(" --auth_type={}", auth_type));
+    }
+
+    let output = Command::new("npm")
+        .arg("login")
+        .arg(args)
+        .stdin(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .output()?;
+
+    if !output.status.success() {
+        let s = String::from_utf8_lossy(&output.stderr);
+        PBAR.error("Registry user account login failed");
```

Comment:
I think it'd be useful to mention the registry they were trying to log into here as well :)

Reply:
Here it seems a little more helpful, because this way we don't depend on the stderr output of npm.